src/controller/middlewares/wrapper/index.js

```diff
@@ -0,0 +1,7 @@
+module.exports = (handler) => async (req, res, next) => {
+  try {
+    return handler(req, res, next);
+  } catch (error) {
+    return next(error);
+  }
+};
```

Comment:
Como uma rota aceita um array de middlewares talvez você possa fazer essa função assim:
```suggestion
const tryCatchMiddlewareWarper =  (handler) => async (req, res, next) => {
  try {
    return handler(req, res, next);
  } catch (error) {
    return next(error);
  }
}
module.exports = (...handlers) => {
if(!handlers.length) throw Error('No middlewares to handle') 
if(handlers.length === 1) return tryCatchMiddlewareWarper(handlers[0]);
return handlers.map(handler => tryCatchMiddlewareWarper(handler))
};
```
E usar assim:
```js
router.get('/',
  wrapper(authentication, admAuthorization, searchAll)
);
```

Reply:
:warning: não testei pra ver se funciona, mas na minha cabeça faz sentido.
Tava procurando um jeito de diminuir a repetição de `warper(middleware)`